src/server.py

```diff
@@ -1,24 +1,59 @@
 # -*- coding: utf-8 -*-
 """Server module to to take HTTP requests."""
+import io
+import mimetypes
+import os
 import socket
 import sys
 import time
+from contextlib import closing
 
-OK_200 = b'200 OK'
-ERR_400 = b'400: Bad Request'
-ERR_405 = b'405: Method Not Allowed'
-ERR_505 = b'505: HTTP Version Not Supported'
 BUFFER_LENGTH = 8
 ADDRINFO = ('127.0.0.1', 5000)
 
-
-# TODO: update response assemble and tests for more sophisticated
-#   testing of each component. Assert for required parts while disrgarding
-#   optional components (but making sure they are in the right place).
-#   split on \r\n\r\n assert there are 2 pieces
-#   for first line of response, use maxsplit arg to split
-#   e.g. "HTTP/1.1 500 Internal Server Error".split(' ', maxsplit=2)
-#   >>> ["HTTP/1.1", "500", "Internal Server Error"]
+PARENT_DIR = os.path.dirname(__file__)
+GRANDPARENT_DIR = os.path.join(PARENT_DIR, '..')
+WEBROOT_PATH = os.path.join(GRANDPARENT_DIR, 'webroot', '')
+
+# Unicode constants for standard re-usable parts.
+CRLF = '\r\n'
+GET = 'GET'
+HTTP1_1 = 'HTTP/1.1'
+HOST = 'Host: '
+CONTENT_TYPE = 'Content-Type: {}'
+CONTENT_LENGTH = 'Content-Length: {}'
+
+TEXT_HTML = 'text/html; charset=utf-8'
+TEXT_PLAIN = 'text/plain; charset=utf-8'
+
+# Easy Reference dictionary with int keys for code/message part.
+HTTP_CODES = {
+    200: '200 OK',
+    400: '400 Bad Request',
+    404: '404 Not Found',
+    405: '405 Method Not Allowed',
+    505: '505 HTTP Version Not Supported',
+}
+
+# updated HTML templates to use unicode strings
+HTML_TEMPLATE = '''
+                <!DOCTYPE html>
+                <html>
+                  <head>
+                    <meta charset="utf-8">
+                    <title></title>
+                  </head>
+                  <body>
+                    <h3>{header}</h3>
+                    <ul>
+                      {list_items}
+                    </ul>
+                  </body>
+                </html>
+                '''
+HTML_LI_TEMPLATE = '''
+                   <li><a href="{full_path}">{short_path}</a></li>
+                   '''
```

Comment:
Nice use of "templating" to solve the problem of building a good HTML representation of your directories.  Well thought out!